fud/fud/stages/verilator/numeric_types.py

```diff
@@ -30,11 +30,19 @@ def __init__(self, value: str, width: int, is_signed: bool):
                 f"The value: {value} of type: "
                 f"{type(value)} should be a non-empty string."
             )
-        if value.startswith("-") and not is_signed:
+        elif value.startswith("-") and not is_signed:
             raise InvalidNumericType(
-                f"A negative value was provided: {value}, " f"and `is_signed` is False."
+                f"A negative value was provided: {value}, and `is_signed` is False."
             )
+        # Some backends may use `x` to represent an uninitialized digit, e.g. `0bxxxx`.
+        # Since this cannot be properly translated into a number, returns error.
         value = value.strip()
+        stripped_prefix = value[2:] if value.startswith("0x") else value
+        if any(digit == "x" for digit in stripped_prefix):
+            raise InvalidNumericType(
+                f"Tried to parse value: {value} "
+                f"with width: {width}, which is uninitialized."
```

Comment:
Can you add to this error message: "This should probably not happen. Please open an issue in the Calyx repository with a minimal reproduction".